src/main/java/mate/academy/lib/BookDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+package mate.academy.lib;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.Book;
+import mate.academy.BookDao;
+import mate.academy.util.ConnectionUtil;
+import mate.academy.util.DataProcessingException;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String INSERT_BOOK_QUERY =
+            "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String SELECT_ALL_BOOKS_QUERY =
+            "SELECT * FROM books";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement preparedStatement =
+                         connection.prepareStatement(INSERT_BOOK_QUERY,
+                                 PreparedStatement.RETURN_GENERATED_KEYS)) {
+                preparedStatement.setString(1, book.getTitle());
+                preparedStatement.setBigDecimal(2, book.getPrice());
+                preparedStatement.executeUpdate();
+                ResultSet resultSet = preparedStatement.getGeneratedKeys();
+                if (resultSet.next()) {
+                    book.setId(resultSet.getLong(1));
+                }
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error creating book", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
```

Comment:
Why is the request in a method here, but in other methods in a constant? Use one method.

P.S. In general, a constant is preferable. But for homework it is better to use it in a method; it is more convenient for us to look at it when there are many methods.